src/sql/src/names.rs

```diff
@@ -2170,6 +2171,44 @@ where
     ResolvedIds(visitor.ids)
 }
 
+#[derive(Debug)]
+pub struct ItemDependencyModifier<'a> {
+    pub modified: bool,
+    pub id_map: &'a BTreeMap<GlobalId, GlobalId>,
+}
+
+impl<'ast, 'a> VisitMut<'ast, Raw> for ItemDependencyModifier<'a> {
+    fn visit_item_name_mut(&mut self, item_name: &mut RawItemName) {
+        if let RawItemName::Id(id, _, _) = item_name {
+            let parsed_id = id.parse::<GlobalId>().unwrap();
+            if let Some(new_id) = self.id_map.get(&parsed_id) {
+                *id = new_id.to_string();
+                self.modified = true;
+            }
+        }
```

Comment:
What happens if we encounter a `RawItemName::Name`? Is that unexpected? If so, we should add an else branch with a panic/assert/unreachable.

Reply:
My thought was that if it was a `RawItemName::Name` then there is no contained id to update. This method is intending only to update ids and assumes the caller takes care of enforcing the name is copied from the item with the current-global-id to the new one

Reply:
Ah ok I see, the name will resolve to the new ID when it's looked up. That LGTM then.